File: app/src/main/java/me/ycdev/android/devtools/security/foo/ParcelableTest.java
package me.ycdev.android.devtools.security.foo;

import android.os.Parcel;
import android.os.Parcelable;

public class ParcelableTest implements Parcelable {
    private int mOomValue;

    public ParcelableTest(int oomValue) {
        mOomValue = oomValue;
    }

    private ParcelableTest(Parcel in) {
        mOomValue = in.readInt();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(mOomValue);
    }

    public static final Parcelable.Creator<ParcelableTest> CREATOR = new
            Parcelable.Creator<ParcelableTest>() {

                public ParcelableTest createFromParcel(Parcel in) {
                    return new ParcelableTest(in);
                }

                public ParcelableTest[] newArray(int size) {
                    return new ParcelableTest[size];
                }

            };
}

File: app/src/main/java/me/ycdev/android/devtools/security/foo/ParcelableTest.kt
package me.ycdev.android.devtools.security.foo

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

class ParcelableTest : Parcelable {
    private var oomValue: Int

    constructor(oomValue: Int) {
        this.oomValue = oomValue
    }

    private constructor(`in`: Parcel) {
        oomValue = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(oomValue)
    }

    companion object {
        @JvmField val CREATOR: Creator<ParcelableTest> =
            object : Creator<ParcelableTest> {
                override fun createFromParcel(`in`: Parcel): ParcelableTest? {
                    return ParcelableTest(`in`)
                }

                override fun newArray(size: Int): Array<ParcelableTest?> {
                    return arrayOfNulls(size)
                }
            }
    }
}